### PR TITLE
add unit tests for api-service condition methods and also refactor out getAPIServiceConditionByType

### PR DESCRIPTION
**Pull request by logicalhan**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
This adds tests around existing functionality. Additionally, this PR refactors out the retrieval of a condition by type into separate method (since the logic is reused in multiple spots) and adds test coverage around the new refactoring.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/sig api-machinery

### CODE REVIEW

File: staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/helpers.go [hunk 1]
```diff
@@ -98,16 +98,21 @@ func NewLocalAvailableAPIServiceCondition() APIServiceCondition {
...
-func SetAPIServiceCondition(apiService *APIService, newCondition APIServiceCondition) {
-	var existingCondition *APIServiceCondition
+// GetAPIServiceConditionByType gets an *APIServiceCondition by APIServiceConditionType if present
+func GetAPIServiceConditionByType(apiService *APIService, conditionType APIServiceConditionType) *APIServiceCondition {
 	for i := range apiService.Status.Conditions {
-		if apiService.Status.Conditions[i].Type == newCondition.Type {
-			existingCondition = &apiService.Status.Conditions[i]
+		if apiService.Status.Conditions[i].Type == conditionType {
+			return &apiService.Status.Conditions[i]
 			break
```
Comment (liggitt) [inline staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/helpers.go]: break isn't needed

Comment (logicalhan) [reply staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/helpers.go]: removed :) thanks for the eyes on this!

**Pull request merged**